flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -369,16 +370,17 @@ protected void cancelJob(
                         deleteClusterDeployment(
                                 deployment.getMetadata(), deploymentStatus, conf, true);
                     }
+                    deploymentStatus.getJobStatus().setState(JobStatus.FINISHED.name());
                     break;
                 case LAST_STATE:
                     deleteClusterDeployment(
                             deployment.getMetadata(), deploymentStatus, conf, false);
+                    deploymentStatus.getJobStatus().setState(JobStatus.FINISHED.name());
                     break;
                 default:
                     throw new RuntimeException("Unsupported upgrade mode " + upgradeMode);
             }
         }
-        deploymentStatus.getJobStatus().setState(JobStatus.FINISHED.name());
```

Comment:
The problem with this and which also complicates the PR is that in some cases the job is already finished / failed or other terminal state in which case no action was taken and we should leave the status as is.

As you said, it's a lot more simpler to simply set the state to "FINISHED" after cancellation :)

Reply:
Because we ran into this problem when we were doing job ops, the cancel job intermediate state always had `FINISHED`, which is not right for our business. So I don't think it's right for the program.

On another note, I'd like to ask what was the reason for doing this in the first place? Is it normal to have an intermediate state finished for cancel requests?

Reply:
The reason is that we wanted to have a single state after executing stop operations. From the operator's perspective it doesn't really matter whether the job was already stopped/failed/cancelled at that point so setting a single FINISHED state is the simplest from the logic perspective.

Reply:
So to reiterate:

1. The current behaviour is the simplest and is correct from the operator's perspective.
2. If users rely on the content of the cr status to tell how exactly the job was stopped, it could be improved. But we should try not to complicate it too much.